### PR TITLE
WIP 655 cyril implement report top filter

### PR DESCRIPTION
**Pull request by cyrielo**
# Description
This PR implements advanced filtering on the report page and moves sidebars filters to the top 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. 

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [ ] Login Screen
- [ ] Single Upload Page
- [ ] Batch Upload Page
- [ ] All Projects Page
- [ ] Sample Details Page
- [ ] Report Page

Specific components 

* 

# Testing Script:

###Login Page
* Successfully login to the application and get taken to the project page

###Single Upload
* Using an AWS path that doesn't exist throws an error
* Successfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

###Batch Upload
* Using an AWS path that doesn't exist throws an error
* Successfully uploaded files from a folder (use test host)
* Navigate to batch upload settings page after upload 
* Host selection from the previous page still selected 
* Uploading a duplicate file throws a warning
* Pipeline gets initiated successfully 

###All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

###Sample Details Page
* Information is consistent from Project Page
* Notes are editable and save successfully 
* Can navigate to report page 

###Report Page
* Data loads successfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:

- [ ] I have run through the testing script to make sure current functionality is unchanged
- [ ] I have done relevant tests that prove my fix is effective or that my feature works
- [ ] I have spent time testing out edge cases for my feature
- [ ] I have updated the test script or pull request template if necessary
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/assets/src/components/PipelineSampleReport.jsx [hunk 1]
```diff
@@ -27,6 +28,9 @@ class PipelineSampleReport extends React.Component {
 
     this.state = {
       taxonomy_details: [],
+      backgroundName: Cookies.get('background_name') || this.report_details.background_model.name,
```
Comment (cdebourcy) [inline app/assets/src/components/PipelineSampleReport.jsx]: New master doesn't have a background_model field in report_details anymore so you may need to refactor a bit

File: app/assets/src/components/PipelineSampleReport.jsx [hunk 2]
```diff
@@ -394,104 +464,130 @@ class PipelineSampleReport extends React.Component {
...
 
-    this.applySearchFilter(0, this.state.excluded_categories, thresholded_taxons);
+  getSavedThresholdFilters() {
+    const activeThresholds = window.localStorage.getItem('activeThresholds');
+    return (activeThresholds) ? JSON.parse(activeThresholds) : [];
+  }
 
-    if (play_animation) {
-      this.flash();
+  applyThresholdFilters(candidateTaxons, rules, animate = true) {
```
Comment (yunfangjuan) [inline app/assets/src/components/PipelineSampleReport.jsx]: I think there are way too many changes than just move the filters to the top.

File: app/assets/src/components/PipelineSampleReport.jsx [hunk 3]
```diff
@@ -55,24 +65,60 @@ class PipelineSampleReport extends React.Component {
...
+        }, {
+          name: 'R log(1/e)',
+          value: 'NR_neglogevalue'
+        }, {
+          name: 'NR %conc',
+          value: 'NR_percentconcordant'
+        }]
+      };
     this.applySearchFilter = this.applySearchFilter.bind(this);
-    this.applyThresholdFilters = this.applyThresholdFilters.bind(this);
```
Comment (yunfangjuan) [inline app/assets/src/components/PipelineSampleReport.jsx]: why do you remove this? This still exists.

Comment (cyrielo) [reply app/assets/src/components/PipelineSampleReport.jsx]: We were binding the `this` context to the method, but we have too many binds already and  we could avoid that by using the arrow functions like this 
`<button onClick={ () => this.methodName() }></button>` then  we don't have to call the bind method again

File: app/assets/src/components/PipelineSampleReport.jsx (hunk 2 above)
Comment (cyrielo) [reply app/assets/src/components/PipelineSampleReport.jsx]: Yeah  because we also had to remove the sidebar and re-style form inputs ui and implement the advanced filtering

**Comment by yunfangjuan** [comment]
A few things observed from the new behavior:
1.  Category filtering: it automatically closes after I click on one of the checkboxes. I think it should let people clicking multiple check boxes without closing. It would only close the category filtering box when the user clicks out of the box. 
2. Scrolling: it doesn't load more results because of the content length. If you download the alpha data, 
http://localhost:3000/samples/18 gives you a good idea. We should make the scrolling work or have a load more link on the bottom. 
3. Advance Filter: it doesn't close when the user clicks outside of the box. Also, we should just use '>=', '<=', nobody is going to use the '==' 
4. Advance filtering doesn't work: if you pick a filter, first time it works. But once you X out the filter, you don't get the results back. The # of results do change but the report section is not rendered. 
5. The search box and advance filtering interacts in an odd way. When a user search something, it shows the filtered search result. But when you open the advance filtering box and pick something, it ignores the search query. It should filter down the search results.  Not showing the results just based on the advance filters. 
6. When you add an advance filter and doesn't click the operand, it should show an error. Or you can make the operand default to '>='
